### ISSUE TITLE
Suggestion: improving visibility of the cursor in the 'm'ap view

### ISSUE THREAD

**Issue by graysage1** [CleverRaven/Cataclysm-DDA#12173]
Right now this happens often when I play: I search for a specific terrain feature in the 'm'ap view, and of course the game finds it right away, and......  I can't see where it is.  Indeed, blue cursor only gives blue background to an ASCII tile, but many game terrain landmarks in and of themselves have the blue background.  For example, abandoned storefront has blue background.  So it becomes trivial to lose your place, as it were, on the 'm'ap.

I wish the cursor was really distinct and obvious so that there'd be no chance of visually losing it among similar looking ASCII tiles.

**Comment by Coolthulhu** [comment]
Not sure what would be good, yet non-intrusive way of marking the current point.

Having a "crosshair" on the entire screen?

**Comment by BevapDin** [comment]
I have the same problem sometimes (and no solution either), but it helps to know the cursor does not move (it's always at the center of the map), only the map moves.

**Comment by John-Candlebury** [comment]
Easiest solution would be to forbid overmap tiles from having the dark blue background. 

Only three buildings use it that I recall:
-Abandoned Store. Used to be  white >
-Police Station. Used to be a yellow >
-Fema Camp Always used blue background

**Comment by Zireael07** [comment]
@John-Candlebury: +1 from me.

Would light blue work? Or cyan?

**Comment by KrazyTheFox** [comment]
What about making it white? I can't recall off-hand any overmap tiles with a white background.

**Comment by graysage1** [comment]
>Having a "crosshair" on the entire screen?

This is probably overkill.  Is there some middle ground?

**Comment by graysage1** [comment]
@John-Candlebury Your idea might work.

**Comment by graysage1** [comment]
>What about making it white? I can't recall off-hand any overmap tiles with a white background.

I believe some tiles have light gray background at least, if not white.  White might be better than what we have now, but without testing it's hard for me to imagine what the contrast level would be.  There is a possibility it will get lost among the light gray backgrounds.

**Comment by graysage1** [comment]
Is it possible to make a cursor that cycles among 6 different colors?  Instead of solid blue, it would cycle every say X ms (using the animation option value maybe) among blue, green, red, etc.  I don't know if I'd want this to be a default though.  If the color change could be made subtle and gradual it might be visible without being annoying, but on a -console build this could be annoying.

Another possibility is to leave the current cursor as is, but add a special key to the 'm'ap view that temporarily highlights the cursor by having a special animations aimed at the cursor being displayed.  So from the right side we'd have < approaching.  From the top we'd have V approaching, and so on.  These characters would approach the cursor location, then disappear, and approach again, in a loop.  This loop would repeat so long as you hold down the "visibility" key.  That way if you know where the cursor is, the cursor can remain subtle, but if you lose it, you can quickly find it via that "show me" key.

**Comment by DavidKeaton** [comment]
How about that?

**Comment by graysage1** [comment]
@DavidKeaton 

Us mortals have to wait for a build to go out before we can test new features.  :)

**Comment by DavidKeaton** [comment]
![untitled](https://cloud.githubusercontent.com/assets/1715605/7335576/1fdbc6ce-eb98-11e4-99f3-4d396980fa5b.png)

**Comment by graysage1** [comment]
I got 0.C-3076 cac8af7 where this is implemented.  It seems good, or better than it was, in terms of visibility.  It's hard to say exactly how much I like it without prolonged play testing.  It looks deserving of being taken for a spin to my eye. :)  I think @BevapDin said he experienced a similar problem with searching on the 'm'ap, so maybe he can provide his opinion too.

If I were the only stakeholder in this issue, I'd be fine with closing this issue for now.

**Comment by DavidKeaton** [comment]
@graysage1 well thank you for those kind words! :-)

**Comment by Coolthulhu** [comment]
It's good, but if it was possible, I think it would be good if it worked more independently of blinking.
If you move it while it is "blinked out", it won't "blink in" until you wait out the blink time. Maybe it could force a blink on every move or (could be less annoying) force a "blink in" on every move?

**Comment by DavidKeaton** [comment]
@Coolthulhu I thought about this too, but I wanted to be able to allow people to disable it via the overlays setting. I'm all for tweaks though, it was just the first thing I thought of when reading this thread. :-)

**Comment by KrazyTheFox** [comment]
Could there be an option to customize the cursor itself (BG/FG color and shape)? For example, I might like to use more of a crosshair shape with a white foreground and magenta background.

**Comment by DavidKeaton** [comment]
@KrazyTheFox There could be, sure, however my fix doesn't go that in depth. Mine is a bit hacky, so if someone has some time to implement this with the new color palette options, that would work well.

**Comment by DavidKeaton** [comment]
@graysage1 If my implementation fixed this issue, you could close and re-open for different improvements to the cursor? :-)

**Comment by graysage1** [comment]
So far I think the cursor is way better than how it used to be.  So I'll gladly close this one.  If someone else has a problem w/ it I guess they can open a new issue or reopen this one.

**Comment by DavidKeaton** [comment]
precisely! and thanks m8: -)
